wen-restart/src/wen_restart.rs

```diff
@@ -871,34 +1043,20 @@ pub fn wait_for_wen_restart(config: WenRestartConfig) -> Result<()> {
                 new_root_slot,
                 new_root_hash,
             } => {
-                if config.cluster_info.id() == config.wen_restart_coordinator {
-                    config
-                        .cluster_info
-                        .push_restart_heaviest_fork(new_root_slot, new_root_hash, 0);
-                    // TODO(wen): remove this aggregate.
-                    aggregate_restart_heaviest_fork(
-                        &config.wen_restart_path,
-                        config.cluster_info.clone(),
-                        config.bank_forks.clone(),
-                        config.exit.clone(),
-                        &mut progress,
-                    )?;
-                    WenRestartProgressInternalState::HeaviestFork {
-                        new_root_slot,
-                        new_root_hash,
-                    }
-                } else {
-                    let (coordinator_slot, coordinator_hash) = receive_restart_heaviest_fork(
-                        config.wen_restart_coordinator,
-                        config.cluster_info.clone(),
-                        config.exit.clone(),
-                        &mut progress,
-                    )?;
-                    // TODO(wen): add verification here.
-                    WenRestartProgressInternalState::HeaviestFork {
-                        new_root_slot: coordinator_slot,
-                        new_root_hash: coordinator_hash,
-                    }
+                let (slot, hash) = send_and_receive_heaviest_fork(
+                    new_root_slot,
+                    new_root_hash,
+                    &config,
+                    &mut progress,
+                    |slot, hash| {
+                        config
+                            .cluster_info
+                            .push_restart_heaviest_fork(slot, hash, 0);
+                    },
+                )?;
+                WenRestartProgressInternalState::HeaviestFork {
+                    new_root_slot: slot,
```

Comment:
side note: just realized it's kind of confusing these are called root slots when they're not really root slots

Reply:
renamed to final_restart_slot, does this sound better?

Reply:
let's keep it consistent and rename it to `my_heaviest_fork_slot`

Reply:
done.